lib/replicator.js

```diff
@@ -662,6 +663,7 @@ class Peer {
 
     if (length === this.remoteLength && fork === this.core.state.fork) {
       this.canUpgrade = upgrade
+      if (upgrade) this.lastUpgradableLength = length
```

Comment:
I think we also need logic for the case where the hypercore gets forked, in which case the `lastUpgradableLength` of the previous fork loses meaning.